plugins/linux/bccbpf/lua/circll.lua

```diff
@@ -0,0 +1,65 @@
+--
+-- BPF circllhist helper
+--
+
+local circll = {}
+
+circll.text = [[
+
+typedef struct {
+  s8 val;
+  s8 exp;
+} circll_bin_t;
+
+#define LLN() if(v > 100) { exp++; v /= 10; } else goto good;
+#define LLN2() LLN() LLN()
+#define LLN4() LLN2() LLN2()
+#define LLN8() LLN4() LLN4()
+#define LLN16() LLN8() LLN8()
+#define LLN32() LLN16() LLN16()
+#define LLN64() LLN32() LLN32()
+#define LLN128() LLN64() LLN64()
+
+static circll_bin_t circll_bin(u64 v, s8 exp_offset) {
+  s8 exp = 1;
+  if(v == 0) return (circll_bin_t)  {.val = 0,    .exp = 0 + exp_offset};
+  if(v < 10) return (circll_bin_t)  {.val = v*10, .exp = 1 + exp_offset};
+  LLN128()
+  if(v > 100) return (circll_bin_t) {.val = -1,   .exp = 0 + exp_offset};
+ good:
+  return (circll_bin_t) {.val = v, .exp = exp + exp_offset};
+}
+]]
```

Comment:
I was not able to get teh exp_offset working, with the bit-shifting tricks used in iolatency.py. So I avoided those by using the bucket struct from libcircllhist. Do you see any drawbacks, @postwait ?